cfme/storage/object_store_object.py

```diff
@@ -104,6 +104,7 @@ class ObjectStoreObject(BaseEntity, WidgetasticTaggable):
         provider: provider
     """
     key = attr.ib()
+    provider = attr.ib(default=None)
```

Comment:
**Question/Required**
Can two object store objects on different providers have the same key?

If so, provider needs to be required, and all() will have to create a provider instance as well.

Reply:
Yup; I got a good idea about it: Using `get_crud_by_name(provider_name)` to create provider and `provider_name` generating from `Cloud Provider Name` or `Storage Manager Name` fetching from table only. It will similar to above filter only difference will return all objects with respective provider. Now no need to set `provider=None`. Improving @mshriver